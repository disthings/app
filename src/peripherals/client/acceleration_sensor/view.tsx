import * as React from "react";
import {View, Text} from "react-native";
import {PeripheralView} from "../../../view/peripheral_view";
import {PeripheralViewProps, PeripheralViewState} from "../../../types";

export class AccelerationSensorView extends PeripheralView<PeripheralViewProps, PeripheralViewState> {

	constructor(props: any, state: any) {
		super(props, state);
	}


	onLayoutChange(_layout: string): void {
		return;
	}

	render(): React.ReactNode {
		let value: string = this.state.data ? JSON.stringify(this.state.data, null, "\t") : "No available data";
		return (<View>
			<Text>Data: {value}</Text>
		</View>);
	}
}