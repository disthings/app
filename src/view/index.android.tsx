import * as React from "react";
import {AppRegistry} from "react-native";
import {ViewContainer} from "./view_container";

class Disthings extends React.Component<object, object> {

  render(): React.ReactNode {
    return (
      <ViewContainer/>
    );
  }
}

AppRegistry.registerComponent("disthings", () => Disthings);
