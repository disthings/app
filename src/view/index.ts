import { AppRegistry } from "react-native";
import {ViewContainer} from "./view_container";

AppRegistry.registerComponent("disthings", () => ViewContainer);