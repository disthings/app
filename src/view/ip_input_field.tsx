import * as React from "react";
import {View, TextInput, Text, TouchableOpacity, StyleSheet} from "react-native";
import {ColorTheme, IPInputFieldProps, IPInputFieldState, IPInputFieldStyle} from "../types";
import {Toast} from "../js_to_typescript_adapters/react_native_root_toast";
import {isValidIPv4} from "../generic_functions";

export class IPInputField<K extends IPInputFieldProps, L extends IPInputFieldState>
	extends React.Component<IPInputFieldProps, IPInputFieldState> {

	private styles: any;

	constructor(props: K, state: L) {
		super(props, state);
		this.state = {
			text: "",
			currentColorTheme: props.currentColorTheme
		};

		props.subscribeToThemeChange((theme: ColorTheme) => {
			this.setState({
				currentColorTheme: theme
			});
		}, "ip_input_field");

		const ipInputFieldStyle: IPInputFieldStyle = this.state.currentColorTheme.ipInputField;

		this.styles = StyleSheet.create({
			view: {
				backgroundColor: ipInputFieldStyle.view.backgroundColor,
				borderWidth: 2,
				borderColor: ipInputFieldStyle.view.borderColor
			},
			button: {
				borderTopWidth: 2,
				padding: 5,
				width: "50%",
				borderColor: ipInputFieldStyle.button.borderColor
			},
			cancelButton: {
				backgroundColor: ipInputFieldStyle.cancelButton.backgroundColor
			},
			okButton: {
				backgroundColor: ipInputFieldStyle.okButton.backgroundColor
			},
			text: {
				fontSize: 16,
				fontWeight: "bold",
				alignSelf: "center"
			},
			textInput: {
				fontSize: 16,
				width: "80%",
				alignSelf: "center"
			},
			buttonContainer: {
				flexDirection: "row"
			}
		});
	}

	render(): React.ReactNode {
		return (<View style={this.styles.view}>
					<Text style={this.styles.text}>Please insert the Server's IP</Text>
					<TextInput style={this.styles.textInput}
							   onChangeText={(newText: string) => { this.setState({text: newText});}}
							   value={this.state.text}
							   autoFocus={true} selectTextOnFocus={true} keyboardType={"numeric"}/>
					<View style={this.styles.buttonContainer}>
						<TouchableOpacity onPress={() => {
							this.props.setNewIP("0.0.0.0");
							Toast.show("No IP-Address inserted");
						}} style={[this.styles.button, this.styles.cancelButton]}>
							<Text style={this.styles.text}>Cancel</Text>
						</TouchableOpacity>
						<TouchableOpacity onPress={() => {
							if(isValidIPv4(this.state.text.trim())) {
								this.props.setNewIP(this.state.text.trim());
							}
							else {
								Toast.show("Invalid IP-Address: " + this.state.text);
							}
						}} style={[this.styles.button, this.styles.okButton]}>
							<Text style={this.styles.text}>Ok</Text>
						</TouchableOpacity>
					</View>
				</View>);
	}
}