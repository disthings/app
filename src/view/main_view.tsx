import * as React from "react";
import {ScrollView, StyleSheet, View} from "react-native";
import {PeripheralTile} from "./peripheral_tile";
import {MainViewProps, MainViewState} from "../types";
import {Peripheral} from "../model/peripheral";

export class MainView<K extends MainViewProps, L extends MainViewState> extends React.Component<MainViewProps, MainViewState> {

	constructor(props: K, state: L) {
		super(props, state);

		this.state = {
			peripherals: this.props.peripherals
		};
	}

	render(): React.ReactNode {

		const peripheralTiles: any = this.state.peripherals.map(item => (
			<PeripheralTile key={(item.peripheral as Peripheral).getName()}
							subscribeToLayoutChange={this.props.subscribeToLayoutChange}
							unsubscribeFromLayoutChange={this.props.unsubscribeFromLayoutChange}
							onPressTile={() => {
								this.props.onPressTile(item.view, item.peripheral as Peripheral);
							}}
							peripheralTileData={item.tile} peripheral={item.peripheral as Peripheral}/>
		));

		return (<ScrollView>
					<View style={styles.view}>
						{peripheralTiles}
					</View>
				</ScrollView>);
	}
}

const styles: any = StyleSheet.create({
	view: {
		flexDirection: "row",
		flex: 1,
		flexWrap: "wrap"
	}
});