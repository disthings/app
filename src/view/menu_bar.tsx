import * as React from "react";
import {TouchableOpacity , StyleSheet, View, Text} from "react-native";
import {MenuBarProps, ViewType} from "../types";

export class MenuBar<K extends MenuBarProps, L extends any> extends React.Component<MenuBarProps, any> {


	constructor(props: K, state: L) {
		super(props, state);
		this.state = {
			currentView: ViewType.MAIN
		};

		props.subscribeOnViewChange((currentView: ViewType) => {
			this.setState({
				currentView: currentView
			});
		});
	}

	render(): React.ReactNode {
		return (
			<View style={style.view}>
				<TouchableOpacity onPress={() => {
					this.props.onPressHomeButton();
					this.setState({
						currentView: ViewType.MAIN
					});
				}} style={style.button}>
					<Text style={this.state.currentView === ViewType.MAIN ? style.selectedButtonText : style.text}>Peripherals</Text>
				</TouchableOpacity>
				<TouchableOpacity onPress={() => {
					this.props.onPressSettingsButton();
					this.setState({
						currentView: ViewType.SETTINGS
					});
				}} style={style.button}>
					<Text style={this.state.currentView === ViewType.SETTINGS ? style.selectedButtonText : style.text}>Settings</Text>
				</TouchableOpacity>
			</View>
		);
	}
}

const style: any = StyleSheet.create({
	view: {
		flexDirection: "row"
	},
	button: {
		backgroundColor: "#a0ffec",
		width: "50%",
		height: 40,
		borderLeftWidth: 1,
		borderRightWidth: 1,
		borderBottomWidth: 4,
		borderColor: "#E6E7E2"
	},
	selectedButtonText: {
		fontWeight: "bold",
		alignSelf: "center",
		margin: 10
	},
	text: {
		alignSelf: "center",
		margin: 10
	}
});