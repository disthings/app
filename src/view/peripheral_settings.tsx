import * as React from "react";

export class PeripheralSettings extends React.Component<any, any> {

}