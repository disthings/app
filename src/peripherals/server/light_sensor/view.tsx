import * as React from "react";
import {View, Text, Button} from "react-native";
import {PeripheralView} from "../../../view/peripheral_view";
import {PeripheralViewState} from "../../../types";

export class LightView extends PeripheralView<any, PeripheralViewState> {

	constructor(props: any, state: any) {
		super(props, state);
	}

	componentDidMount(): void {
		this.setState({
			status: {isOn: true}
		});
	}

	render(): React.ReactNode {

		let value: string = this.state.data ? JSON.stringify(this.state.data, null, "\t") : "No available data";

		return (<View>
			<Text>Level: {value}</Text>
			<Button onPress={() => {
				this.sendCommand({
					commandName: this.state.status.isOn ? "off" : "on",
					commandData: {}
				});
				this.setState({
					status: {
						isOn: !this.state.status.isOn
					}
				});
			}}
				title={this.state.status.isOn ? "off" : "on"}/>
		</View>);
	}
}
