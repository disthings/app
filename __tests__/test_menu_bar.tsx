import "react-native";
import * as React from "react";
import {MenuBar} from "../src/view/menu_bar";

// Note: test renderer must be required after react-native.
import * as renderer from "react-test-renderer";


test("renders correctly", () => {
	const tree = renderer.create(<MenuBar
		onPressHomeButton={() => {console.log("HOME");}}
		onPressSettingsButton={() => {console.log("SETTINGS");}}
		subscribeOnViewChange={() => {console.log("VIEW_CHANGE");}}
	/>).toJSON();
	expect(tree).toMatchSnapshot();
});