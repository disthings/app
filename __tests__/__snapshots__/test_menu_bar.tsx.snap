// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#a0ffec",
        "borderBottomWidth": 4,
        "borderColor": "#E6E7E2",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "height": 40,
        "opacity": 1,
        "width": "50%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontWeight": "bold",
          "margin": 10,
        }
      }
    >
      Peripherals
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#a0ffec",
        "borderBottomWidth": 4,
        "borderColor": "#E6E7E2",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "height": 40,
        "opacity": 1,
        "width": "50%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "margin": 10,
        }
      }
    >
      Settings
    </Text>
  </View>
</View>
`;
